src/excmds.ts

```diff
@@ -2500,6 +2501,16 @@ export async function undo(item = "recent"): Promise<number> {
             browser.sessions.restore(lastSession.tab.sessionId)
             return lastSession.tab.id
         }
+    } else if (item === "tab_strict") {
+        const lastSession = sessions.find(s => {
```

Comment:
Thanks for this PR!

I think we have a bit too much code repeated here now. Would you mind refactoring it so that the if statement set a filter function (i.e. the s => bit) and then move the actual filtering and restoration out of the ifs?

If not I'll get round to it in a few days time : )

Reply:
Agreed. I simplified it quite a lot. Were you thinking something along the line of this?